### PR TITLE
CI: Add Circle tests

### PR DESCRIPTION
**Pull request by effigies**
This is a basic replication of the CircleCI tests from [nipy/nipype](https://github.com/nipy/nipype/). Eventually this should be moved into `recipe.yml`, once we figure out what that looks like. I'm using a pinned `nipype/nipype:base` Docker image.

### CODE REVIEW

**Comment by effigies** [comment]
Hmm. Looks like I'll need to merge a Circle config to master to kick things off. We'll see how this goes.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    working_directory: /home/circleci/niflow
+    steps:
+      - checkout:
+          path: /home/circleci/niflow
+      - run: *set_pr_number
+      - run: *generate_dockerfiles
+      - run: *modify_nipype_version
+      - run: *get_base_image
+      - run: *build_main_image_py36
+      - run: *build_main_image_py27
```
Comment (kaczmarj) [inline .circleci/config.yml:152]: Should we test python 2.7?

**Comment by kaczmarj** [comment]
Other than my comment about whether to test Python 2.7, this looks fine to me.

**Comment by effigies** [comment]
Yeah, right now I'm just copying directly from nipype. I think there was discussion of wanting to re-import these workflows into 1.x, for backwards compatibility, so I'm sticking with the current nipype tests.

But we should figure out how to write a `recipe.yml` where we can specify Python versions to run on, and we can drop the manually-written `config.yml` altogether.

**Pull request merged**